From 7fc3b93309d04237440d3c9a142d46ec40c4c9c1 Mon Sep 17 00:00:00 2001
From: imititel <ionel-catalin.mititelu@intel.com>
Date: Thu, 26 Mar 2020 11:02:03 +0200
Subject: [PATCH 2/2] Split HSM & IFSI into separate IOMMU groups
Patch-mainline: Never, downstream fix.
References: jsc#EB-390

In order for HSM & IFSI to work in parallel
split the two functions up into separate IOMMU groups.

We have agreed that the change is tested for specific 
workload so potential harm should be negligible. 
But it is a non-upstream (and likely not upstreamable solution) 
and we cannot really carry it for ever so a different approach 
has to be adopted for future products based on a different base kernel.

Elektrobit is responsible for this patch.

Signed-off-by: Bogdan Lezhepekov <blezhepekov@suse.de>
Signed-off-by: Peter Kunath <Peter.Kunath@elektrobit.com>
Signed-off-by: Muhammad Faisal <Muhammad.Faisal@elektrobit.com>

---
 drivers/pci/quirks.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -5014,6 +5014,8 @@ static const struct pci_dev_acs_enabled
 	{ PCI_VENDOR_ID_NXP, 0x8d9b, pci_quirk_nxp_rp_acs },
 	/* Zhaoxin Root/Downstream Ports */
 	{ PCI_VENDOR_ID_ZHAOXIN, PCI_ANY_ID, pci_quirk_zhaoxin_pcie_ports_acs },
+	{ PCI_VENDOR_ID_INTEL, 0x19e5, pci_quirk_mf_endpoint_acs },
+	{ PCI_VENDOR_ID_INTEL, 0x19e6, pci_quirk_mf_endpoint_acs },
 	{ 0 }
 };
 
